packages/cosmos_ui_components/lib/components/template/cosmos_action_sheet.dart

```diff
@@ -0,0 +1,156 @@
+import 'dart:io';
+
+import 'package:cosmos_ui_components/cosmos_text_theme.dart';
+import 'package:cosmos_ui_components/cosmos_theme.dart';
+import 'package:flutter/cupertino.dart';
+import 'package:flutter/material.dart';
+
+Future<T?> showCosmosActionSheet<T>({
+  required BuildContext context,
+  Widget? title,
+  required List<CosmosActionSheetItem> actions,
+  VoidCallback? cancelAction,
+}) async {
+  return _show<T>(
+    context,
+    title,
+    actions,
+    cancelAction,
+  );
+}
+
+Future<T?> _show<T>(
+  BuildContext context,
+  Widget? title,
+  List<CosmosActionSheetItem> actions,
+  VoidCallback? cancelAction,
+) {
+  if (Platform.isIOS) {
+    return _showCupertinoBottomSheet(
+      context,
+      title,
+      actions,
+      cancelAction,
+    );
+  } else {
+    return _showMaterialBottomSheet(
+      context,
+      title,
+      actions,
+      cancelAction,
+    );
+  }
+}
+
+Future<T?> _showCupertinoBottomSheet<T>(
+  BuildContext context,
+  Widget? title,
+  List<CosmosActionSheetItem> actions,
+  VoidCallback? cancelAction,
+) {
+  final defaultTextStyle = CosmosTextTheme.actionSheet;
+  return showCupertinoModalPopup(
+    context: context,
+    builder: (BuildContext buildContext) {
+      return CupertinoActionSheet(
+        title: title,
+        actions: actions.map((action) {
+          return Material(
+            color: Colors.transparent,
+            child: CupertinoActionSheetAction(
+              onPressed: action.onPressed,
+              child: Text(
+                action.text,
+                style: defaultTextStyle.copyWith(color: action.isCriticalAction ? Colors.red : Colors.lightBlue),
+              ),
+            ),
+          );
+        }).toList(),
+        cancelButton: CupertinoActionSheetAction(
+          onPressed: cancelAction ?? () => Navigator.of(buildContext).pop(),
+          child: DefaultTextStyle(
+            style: defaultTextStyle.copyWith(color: Colors.lightBlue),
+            textAlign: TextAlign.center,
+            child: const Text('Cancel'),
+          ),
+        ),
+      );
+    },
+  );
+}
+
+Future<T?> _showMaterialBottomSheet<T>(
+  BuildContext context,
+  Widget? title,
+  List<CosmosActionSheetItem> actions,
+  VoidCallback? cancelAction,
+) {
+  final defaultTextStyle = CosmosTextTheme.actionSheet;
+  return showModalBottomSheet<T>(
+    context: context,
+    elevation: 0,
+    isScrollControlled: true,
+    shape: const RoundedRectangleBorder(
+      borderRadius: BorderRadius.only(
+        topLeft: Radius.circular(30),
+        topRight: Radius.circular(30),
+      ),
+    ),
+    builder: (BuildContext buildContext) {
+      final screenHeight = MediaQuery.of(context).size.height;
+      return ConstrainedBox(
+        constraints: BoxConstraints(
+          maxHeight: screenHeight - (screenHeight / 10),
+        ),
```

Comment:
this logic of ConstrainedBox is very neat and could be moved into the `CosmosBottomSheetContainer`

Reply:
Updated